src/mlpack/methods/decision_tree/decision_tree_regressor_impl.hpp

```diff
@@ -0,0 +1,1073 @@
+    MatType data,
+    const data::DatasetInfo& datasetInfo,
+    ResponsesType responses,
+    WeightsType weights,
+    const size_t minimumLeafSize,
+    const double minimumGainSplit,
+    const size_t maximumDepth,
+    DimensionSelectionType dimensionSelector,
+    const std::enable_if_t<arma::is_arma_type<
+        typename std::remove_reference<WeightsType>::type>::value>*)
+{
+  using TrueMatType = typename std::decay<MatType>::type;
+  using TrueResponsesType = typename std::decay<ResponsesType>::type;
+  using TrueWeightsType = typename std::decay<WeightsType>::type;
+
+  TrueMatType tmpData(std::move(data));
+  TrueResponsesType tmpResponses(std::move(responses));
+  TrueWeightsType tmpWeights(std::move(weights));
+
+  // Set the correct dimensionality for the dimension selector.
+  dimensionSelector.Dimensions() = tmpData.n_rows;
+
+  // Pass off work to the weighted Train() method.
+  Train<true>(tmpData, 0, tmpData.n_cols, datasetInfo, tmpResponses,
+      tmpWeights, minimumLeafSize, minimumGainSplit, maximumDepth,
+      dimensionSelector);
+}
+
+//! Construct and train on numeric data with weights.
+template<typename FitnessFunction,
+         template<typename> class NumericSplitType,
+         template<typename> class CategoricalSplitType,
+         typename DimensionSelectionType,
+         bool NoRecursion>
+template<typename MatType, typename ResponsesType, typename WeightsType>
+DecisionTreeRegressor<FitnessFunction,
+                      NumericSplitType,
+                      CategoricalSplitType,
+                      DimensionSelectionType,
+                      NoRecursion>::DecisionTreeRegressor(
+    MatType data,
+    ResponsesType responses,
+    WeightsType weights,
+    const size_t minimumLeafSize,
+    const double minimumGainSplit,
+    const size_t maximumDepth,
+    DimensionSelectionType dimensionSelector,
+    const std::enable_if_t<
+        arma::is_arma_type<
+        typename std::remove_reference<
+        WeightsType>::type>::value>*)
+{
+  using TrueMatType = typename std::decay<MatType>::type;
+  using TrueResponsesType = typename std::decay<ResponsesType>::type;
+  using TrueWeightsType = typename std::decay<WeightsType>::type;
+
+  // Copy or move data.
+  TrueMatType tmpData(std::move(data));
+  TrueResponsesType tmpResponses(std::move(responses));
+  TrueWeightsType tmpWeights(std::move(weights));
+
+  // Set the correct dimensionality for the dimension selector.
+  dimensionSelector.Dimensions() = tmpData.n_rows;
+
+  // Pass off work to the weighted Train() method.
+  Train<true>(tmpData, 0, tmpData.n_cols, tmpResponses, tmpWeights,
+      minimumLeafSize, minimumGainSplit, maximumDepth, dimensionSelector);
+}
+
+//! Take ownership of another tree and train with weights.
+template<typename FitnessFunction,
+        template<typename> class NumericSplitType,
+        template<typename> class CategoricalSplitType,
+        typename DimensionSelectionType,
+        bool NoRecursion>
```

Comment:
```suggestion
         template<typename> class NumericSplitType,
         template<typename> class CategoricalSplitType,
         typename DimensionSelectionType,
         bool NoRecursion>
```
Tiny spacing fix. :)